src/mon/OSDMonitor.cc

```diff
@@ -3205,18 +3205,15 @@ void OSDMonitor::update_creating_pgs()
 		     << " acting_primary:" << last_acting_primary
 		     << " -> " << acting_primary << dendl;
 	    // note epoch if the target of the create message changed.
-	    // creating_pgs is updated here instead of in
-	    // scan_for_creating_pgs() because we don't have the updated pg
-	    // mapping by then.
-	    created = mapping.get_epoch();
+	    mapped = mapping.get_epoch();
```

Comment:
I think here we also need to remove the pg from it's old position.. pgs_by_epoch.second.erase(pgid)?